cmd/sonobuoy/app/images.go

```diff
@@ -19,45 +19,108 @@ package app
 import (
 	"fmt"
 	"os"
+	"syscall"
 
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
 	"github.com/heptio/sonobuoy/pkg/errlog"
 	"github.com/heptio/sonobuoy/pkg/image"
 	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
-	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+	"golang.org/x/crypto/ssh/terminal"
+	"golang.org/x/net/context"
 )
 
 var imagesflags imagesFlags
 
-type imagesFlags struct {
-	e2eRegistryConfig string
-	plugin            string
-	kubeconfig        Kubeconfig
-
-	imagesflags *pflag.FlagSet
-}
+const dockerClientVersion = "1.37"
```

Comment:
Do you know how much support above/below this there is?

Reply:
Honestly no, I'm spinning up a windows machine to validate this works there as well.

Reply:
I commented at the place the err hit me; but this does seem like it will cause at least some problems. I couldnt download images even though a pull worked. On version 1.39 locally.

Reply:
I think we need to think about this particular point. Maybe there is a good way to grab the local version and it would generally work if they aren't really breaking their api much. Hopefully we dont need this to be manually fed in; maybe we can even try shelling out and checking it? That would also ensure the docker daemon is running.

Reply:
If you could pull then seems like the version issue is not the problem. Are you sure you got all the images downloaded? It took me a good 10mins to gen the tar file.

Reply:
I also confirmed that when modified it to 1.39 the download worked for me without other code modifications.

Reply:
Take out the version all together, does it work?

Reply:
Surprisingly yes it seems to. It is taking a while to run but in the error case it failed right away. Just takes a while to download 7g. *heads to look at the docker docs

Reply:
It worked for a while for me without. Then at some point, it started to error. From my understanding, we're specifying the version of the api that we want.

Reply:
Looks like we should just be using `cli, err := client.NewClientWithOpts(client.FromEnv)` which will handle defaults and also allow overrides for users to specify their versions if there are issues.

Reply:
I double checked how kind did it and they shell out to the local client. May switch over to that which also removes the need for any type of auth: https://github.com/kubernetes-sigs/kind/blob/master/pkg/container/docker/pull.go#L30:6